api/webhooks/views.py

```diff
@@ -1,302 +1,180 @@
+        serializer = WebhookOrderCancelSerializer(data=request.data)
+        if not serializer.is_valid():
+            sentry_webhook_error(serializer.errors)
+            return Response(data=serializer.errors, status=400)
+
+        order_id: uuid.UUID = serializer.validated_data["orderId"]
+
+        payment_order: PaymentOrder | None = PaymentOrder.objects.filter(remote_id=order_id).first()
+        if payment_order is None:
+            msg = f"Payment order {order_id=!s} not found"
+            capture_message(msg)
+            return Response(data={"message": msg}, status=404)
+
+        if payment_order.status not in [OrderStatus.DRAFT]:
+            msg = "Order is already in a state where no updates are needed"
+            return Response(data={"message": msg}, status=200)
+
         try:
-            self.validate_request(request)
-
-            remote_id = request.data.get("orderId", "")
-            payment_order = PaymentOrder.objects.filter(remote_id=remote_id).first()
-            if not payment_order:
-                raise WebhookError(message="Order not found", status_code=404)
-
-            # Order is already in a state where no updates are needed
-            if payment_order.status != OrderStatus.DRAFT:
-                return Response(status=200)
-
-            # Check order status from the API
-            order = get_order(remote_id, payment_order.reservation.user.id)
-            order_status = getattr(order, "status", "payment_not_found")
-
-            if order_status != "cancelled":
-                capture_message(
-                    f"Received order cancellation webhook for order {remote_id} "
-                    f"that is not in cancelled state: {order_status}",
-                    level="warning",
-                )
-                raise WebhookError(message="Invalid order state", status_code=400)
-
-            payment_order.status = OrderStatus.CANCELLED
-            payment_order.processed_at = datetime.now().astimezone(get_default_timezone())
-            payment_order.save()
-
-            return Response(status=200)
-        except WebhookError as err:
-            with push_scope() as scope:
-                scope.set_extra("details", "Invalid order webhook")
-                scope.set_extra("data", err.to_json())
-                scope.set_extra("status_code", err.status_code)
-                capture_exception(err)
-            return Response(data=err.to_json(), status=err.status_code)
-        except GetOrderError as err:
-            with push_scope() as scope:
-                scope.set_extra("details", "Order checking failed")
-                scope.set_extra("remote_id", remote_id)
-                capture_exception(err)
-            return Response(
-                data={"status": 500, "message": "Problem with upstream service"},
-                status=500,
-            )
+            order = get_order(order_id)
```

Comment:
I guess this takes care of https://helsinkisolutionoffice.atlassian.net/browse/TILA-2912

Reply:
Yes.